FESOM2 standalone simulation from documentation

I need to do a fesom standalone simulation for a new mesh representing the mid-Pliocene ocean conditions. This issue is to update on errors encountered while following the [documentation](https://fesom2.readthedocs.io/en/latest/getting_started/getting_started.html), as discussed with @koldunovn, @pgierz, @patrickscholz, and Sesh (I don't know his username).

Machine: `levante`
Branch: `refactoring`
Mesh: `/work/ab0246/a270179/runtime/awicm3-v3.1/input/fesom2/midpli/`
Model directory: `/work/ab0246/a270179/runtime/awicm3-v3.1/model_codes/fesom-standalone/fesom2/`

**First step: build model executable**

- To do the mesh partitioning, one has to switch from `master` to `refactoring` branch, otherwise it does not update the available machines for submitting the job. 

After doing this, the mesh partitioning was easy to perform following documentation.

**Second step: running the model**

- My mesh is of mid-Pliocene conditions based on COREII. Is there something to be done prior to running the model? Which naming should I use, `core2` or `midpli`?
- I created a link on `/work/ab0246/a270179/runtime/awicm3-v3.1/model_codes/fesom-standalone/fesom2/results/` for results and created a `fesom.clock` there using the start year of 1990 instead of 1958. Is that ok?
- changes made to `namelist.config`:
```
&timestep
step_per_day=32 !96 !96 !72 !72 !45 !72 !96
run_length=10 !62 !62 !62 !28
run_length_unit='y'             ! y, m, d, s
/

&clockinit              ! the model starts at
timenew=0.0
daynew=1
yearnew=1990
/

&paths
MeshPath='/work/ab0246/a270179/runtime/awicm3-v3.1/input/fesom2/midpli/'
ClimateDataPath='/pool/data/AWICM/FESOM2/INITIAL/phc3.0/'
ResultPath='/work/ab0246/a270179/runtime/awicm3-v3.1/experiments_testing/fesom-standalone/'
/
```
***Notes:*** there should be more description on having to define a result path in `namelist.config`, and of what is exactly the run length.

- changes made to `/work/ab0246/a270179/runtime/awicm3-v3.1/model_codes/fesom-standalone/fesom2/work/job_levante`
```
#SBATCH --job-name=midpli_test1
#SBATCH -p compute
#SBATCH --ntasks-per-node=108
#SBATCH --ntasks=432
#SBATCH --time=00:30:00
#SBATCH -o slurm-out.out
#SBATCH -e slurm-err.out
#SBATCH -A ab0246
```
***Notes:*** None of the above-mentioned lines are described in the documentation, except the number of tasks, which depends on the distribution chosen. 


It does not run and slurm messages and other files are kept in the work folder, instead of in the results folder. Is it ok?
Error message in `slurm-err.out`:
```
/var/spool/slurmd/job6868133/slurm_script: line 14: /work/ab0246/a270179/runtime/awicm3-v3.1/model_codes/fesom_standalone/fesom2/env/levante.dkrz.de/shell.intel: No such file or directory
217: fesom.x: error while loading shared libraries: libmkl_intel_lp64.so.2: cannot open shared object file: No such file or directory
```

> looks like you download FESOM with esm-tools. For "clear experiment" it would be probably better to clone it directly from repo?

Hi, I did try with esm-tools before, but inside `/work/ab0246/a270179/runtime/awicm3-v3.1/model_codes/fesom_standalone/fesom2` I was trying the clear experiment, cloning from the repo and following the instructions. 

As I wanted to fix the typo shown by @patrickscholz, I deleted the repo and tried to clone and compile the model again. The compilation now did not work with the message: 

```
ld: cannot find -lFALSE
make[2]: *** [src/CMakeFiles/fesom.dir/build.make:1642: src/fesom] Error 1
make[2]: Leaving directory '/work/ab0246/a270179/runtime/awicm3-v3.1/model_codes/fesom2/build'
make[1]: *** [CMakeFiles/Makefile2:140: src/CMakeFiles/fesom.dir/all] Error 2
make[1]: Leaving directory '/work/ab0246/a270179/runtime/awicm3-v3.1/model_codes/fesom2/build'
make: *** [Makefile:139: all] Error 2
```

In this way, `fesom.x` is not created and I can't try to run the model again. My only idea is that simply deleting the folder does not do the trick, is there a better way to uninstall the model and try clean again?

--- a/lib/parms/CMakeLists.txt
+++ b/lib/parms/CMakeLists.txt
@@ -6,7 +6,10 @@ project(parms C)
 set(src_home ${CMAKE_CURRENT_LIST_DIR}) # path to src directory starting from the dir containing our CMakeLists.txt
 file(GLOB all_sources ${src_home}/src/*.c ${src_home}/src/DDPQ/*.c)
 
-include("${CMAKE_CURRENT_LIST_DIR}/../../cmake/FindBLAS.cmake")
+find_package(BLAS) # standard way to find blas
+if( NOT BLAS_FOUND) # try older way to find blas
+   include("${CMAKE_CURRENT_LIST_DIR}/../../cmake/FindBLAS.cmake")
+endif()
 
 # create our library (set its name to name of this project)
 if(${BUILD_FESOM_AS_LIBRARY})
@@ -16,11 +19,13 @@ else()
 endif()
 
 if(${BLAS_FOUND})
-target_compile_definitions(${PROJECT_NAME} PRIVATE PARMS USE_MPI REAL=double DBL FORTRAN_UNDERSCORE VOID_POINTER_SIZE_8 HAS_BLAS)
-target_link_libraries(${PROJECT_NAME} INTERFACE ${BLAS_C_LIBRARIES} $ENV{UBUNTU_BLAS_LIBRARY})
+  message("BLAS FOUND ---------------------------")
+  target_compile_definitions(${PROJECT_NAME} PRIVATE PARMS USE_MPI REAL=double DBL FORTRAN_UNDERSCORE VOID_POINTER_SIZE_8 HAS_BLAS)
+  target_link_libraries(${PROJECT_NAME} INTERFACE ${BLAS_C_LIBRARIES} $ENV{UBUNTU_BLAS_LIBRARY})
 else()
-target_compile_definitions(${PROJECT_NAME} PRIVATE PARMS USE_MPI REAL=double DBL FORTRAN_UNDERSCORE VOID_POINTER_SIZE_8)
-target_link_libraries(${PROJECT_NAME} INTERFACE $ENV{UBUNTU_BLAS_LIBRARY})
+  message("BLAS NOT FOUND ***********************")
+  target_compile_definitions(${PROJECT_NAME} PRIVATE PARMS USE_MPI REAL=double DBL FORTRAN_UNDERSCORE VOID_POINTER_SIZE_8)
+  target_link_libraries(${PROJECT_NAME} INTERFACE $ENV{UBUNTU_BLAS_LIBRARY})
 endif()
 
 target_include_directories(${PROJECT_NAME}
@@ -28,7 +33,8 @@ target_include_directories(${PROJECT_NAME}
 	INTERFACE ${src_home}/src/../include
 )
 if(${CMAKE_C_COMPILER_ID} STREQUAL "Intel")
-	target_compile_options(${PROJECT_NAME} PRIVATE -no-prec-div -no-prec-sqrt -fast-transcendentals -fp-model precise)
+   target_compile_options(${PROJECT_NAME} PRIVATE -no-prec-div -no-prec-sqrt -fast-transcendentals -fp-model precise)
+   
    if(${FESOM_PLATFORM_STRATEGY} STREQUAL  levante.dkrz.de )
       target_compile_options(${PROJECT_NAME} PRIVATE -march=core-avx2 -mtune=core-avx2)
    endif()

--- a/lib/parms/CMakeLists.txt
+++ b/lib/parms/CMakeLists.txt
@@ -14,12 +14,19 @@ if(${BUILD_FESOM_AS_LIBRARY})
 else()
   add_library(${PROJECT_NAME} ${all_sources})
 endif()
+
+if(${BLAS_FOUND})
 target_compile_definitions(${PROJECT_NAME} PRIVATE PARMS USE_MPI REAL=double DBL FORTRAN_UNDERSCORE VOID_POINTER_SIZE_8 HAS_BLAS)
+target_link_libraries(${PROJECT_NAME} INTERFACE ${BLAS_C_LIBRARIES} $ENV{UBUNTU_BLAS_LIBRARY})
+else()
+target_compile_definitions(${PROJECT_NAME} PRIVATE PARMS USE_MPI REAL=double DBL FORTRAN_UNDERSCORE VOID_POINTER_SIZE_8)
+target_link_libraries(${PROJECT_NAME} INTERFACE $ENV{UBUNTU_BLAS_LIBRARY})
+endif()
+
 target_include_directories(${PROJECT_NAME}
 	PRIVATE ${src_home}/src/../include ${src_home}/src/include
 	INTERFACE ${src_home}/src/../include
 )
-target_link_libraries(${PROJECT_NAME} INTERFACE ${BLAS_C_LIBRARIES} $ENV{UBUNTU_BLAS_LIBRARY})
 if(${CMAKE_C_COMPILER_ID} STREQUAL "Intel")
 	target_compile_options(${PROJECT_NAME} PRIVATE -no-prec-div -no-prec-sqrt -fast-transcendentals -fp-model precise)
    if(${FESOM_PLATFORM_STRATEGY} STREQUAL  levante.dkrz.de )

